pyinteraph/reformat_interaction_network_graph.py

```diff
@@ -0,0 +1,97 @@
+import sys
+
+import argparse
+import functools
+import logging
+import networkx as nx
+import numpy as np
+import MDAnalysis as mda
+import warnings
+
+
+from pyinteraph.file_validation import ArgumentParserFileExtensionValidation
+
+warnings.filterwarnings("ignore")
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="[%(asctime)s] %(levelname)s | %(message)s",
+    datefmt="%Y/%m/%d %H:%M:%S"
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ReformatDatGraph:
+    def __init__(self, interaction_network_file, output_name, reference_structure_file=None):
+        self.reference_structure_file = reference_structure_file
+        self.interaction_network_file = interaction_network_file
+        self.output_name = output_name
+
+    @property
+    @functools.lru_cache()
+    def interaction_network(self):
+        try:
+            interaction_network = np.loadtxt(self.interaction_network_file)
+            return interaction_network
+        except:
+            sys.exit(1)
+
+    @property
+    @functools.lru_cache()
+    def reference_structure(self):
+        if not self.reference_structure_file:
+            return
+        try:
+            return mda.Universe(self.reference_structure_file)
+        except:
+            logger.warning("Reference structure not readable.")
+            return
+
+    @property
+    @functools.lru_cache()
+    def node_names(self):
+        if not self.reference_structure:
+            logger.warning("Auto-generated numbers will be used to rename nodes.")
+            return [str(i) for i in range(1, self.interaction_network.shape[0] + 1)]
+        return list(map(lambda r: f"{r.segment.segid}-{r.resnum}{r.resname}", self.reference_structure.residues))
+
+    @property
+    def interaction_network_graph(self):
+        try:
+            uploaded_interaction_network_graph = nx.Graph(self.interaction_network)
```

Comment:
I don't think we need to call this variable `uploaded` - uploaded in general refers to upload through a network

Reply:
also I think this `try` clause should end here and exit with error if you can't load the graph. Otherwise you won't catch other errors later (e.g. if the graph and PDB don't have the same number of residues)